source/server/listener_impl.h

```diff
@@ -293,9 +324,30 @@ class ListenerImpl final : public Network::ListenerConfig,
   void createUdpListenerFilterChain(Network::UdpListenerFilterManager& udp_listener,
                                     Network::UdpReadFilterCallbacks& callbacks) override;
 
+  void setStopped() { is_stopped_ = true; }
```

Comment:
ASSERT(!is_stopped_)? Or can a listener be stopped multiple times?

Reply:
I will be surprised if stop is not idempotent. Will give a try